### ISSUE TITLE
Dropout active during inference and evaluation?

### ISSUE THREAD

**Issue by lukkch** [ipolharvard/ethos-ares#20]
Hi!

As i don't see a model.eval() call anywhere in the repo, i'm pretty sure the dropout units are active during inference as well as during loss estimation while training.

The @torch.inference_mode() and @torch.no_grad() decorators only disable gradient computation/reduce resource utilization, and don't set the model to evaluation mode.

I just stumbled upon this so I'm not sure what the implications of this are and haven't done any experiments to compare, just wanted to open the discussion for now.

Best
Lukas

**Comment by prenc** [comment]
Hi @lukkch,
Thanks for you interest in our work.

Good catch! This is definitely a bug. The code in this repo was used to generate the results reported in [the ETHOS-ARES paper](https://academic.oup.com/gigascience/article/doi/10.1093/gigascience/giaf107/8268904).

Here you can find a summary of most downstream tasks after fixing this and several other issues: [spreadsheet](https://docs.google.com/spreadsheets/d/198wdPCYDPYflkKfc3wuS68lO6Ilec4IrZI3F_85RdXw/edit?gid=0#gid=0).

I hope this helps, let me know if you have any more questions.

**Comment by lukkch** [comment]
Hi @prenc,

Thanks for the quick reply and thanks for the spreadsheet, very interesting! 

Do you have any (rough) numbers on how much increasing the rep_num from 20 to 100 improved downstream performance?

**Comment by prenc** [comment]
@lukkch I'm so sorry what the late reply, it depends on the task, but typically it shouldn't be very large from 1-10%.
